lib/crowbar/client/command/proposal/edit.rb

```diff
@@ -140,6 +150,34 @@ def payload_content
               from_editor
             end
           end
+
+          def check_nodes(configuration)
+            invalid_nodes = []
+
+            configuration["deployment"].each do |service_name, service|
+              next if service["elements"].nil?
+
+              service["elements"].each do |role, nodes|
+                nodes.each do |node|
+                  unless valid_elements.include?(node)
+                    invalid_nodes << { "node"=> node, "role"=> role, "service_name"=> service_name }
+                  end
+                end
+              end
+            end
+
+            return if invalid_nodes.empty?
+
+            error_str = ""
+
+            invalid_nodes.each do |error|
+              error_str += "ERROR : #{error["node"]} is not available in the list of" \
+              " possible nodes and clusters, or is invalid IP. Added in role #{error["role"]}" \
```

Comment:
I don't think using IPs is supported in the elements lists. AFAIK only node/cluster names are allowed.